src/util.rs

```diff
@@ -55,7 +55,9 @@ impl Fairing for AppHeaders {
         res.set_raw_header("Referrer-Policy", "same-origin");
         res.set_raw_header("X-Content-Type-Options", "nosniff");
         res.set_raw_header("X-Robots-Tag", "noindex, nofollow");
-        res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
+        if !(req_uri_path.starts_with("/icons/") && req_uri_path.ends_with("/icon.png")) {
+            res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
+        }
```

Comment:
I'd go with the same process as above (might even be slightly more performant):

```suggestion
        res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
        if req_uri_path.ends_with("/icon.png") {
            res.remove_header("Cross-Origin-Resource-Policy");
        }
```

Reply:
How would this be more performant?
You first always set the header, then always check if it ends on a specific string.
While you can also negate the check if it always ends on a specific string and then just set the header.

Reply:
Why would it be more performant to set a header and then potentially remove it? Seems like it would be simplest just to do
```
if !req_uri_path.ends_with("/icon.png") {
    res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
}
```

Reply:
branch prediction. but maybe not anymore. maybe the compilers are much better these days. I am probably too old....

Reply:
also, I have noticed that in some projects clippy will complain because clippy prefers matches over not matches. But maybe that's not the case here either

Reply:
The clippy lint gets triggered if there would have been a function `not_ends_with()` which makes it more clear because missing the `!` in front is easier to do, and a function with that name is more descriptive.

And, doesn't branch prediction work the same way for the exclusion? If the CPU can predict when to not call it, that is fine too ;)